sudo/src/run_handler.rs

```diff
@@ -191,6 +191,53 @@ pub fn run_target(
     do_request(req, copy_env, manually_requested_dir)
 }
 
+/// Returns all the extensions from PATHEXT that are used for "executables"
+fn get_path_extensions() -> Vec<String> {
+    env::var("PATHEXT")
+        .unwrap_or_else(|_| {
+            // If PATHEXT isn't set, use the default
+            ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC".to_string()
+        })
+        .split(';')
+        .map(|ext| ext.to_string())
+        .collect()
+}
+
+/// Searches the PATH for the given filename, with the extensions from PATHEXT.
+/// Returns the full path to the file if it's found, or an error if it's not.
+fn search_path_with_extensions(filename: &str) -> Result<String> {
+    let filename = HSTRING::from(filename);
+    let mut buffer = vec![0; 260];
+    let extensions = get_path_extensions();
+    for extension in extensions {
+        let extension_hstring = HSTRING::from(extension);
+        // SearchPathW will ignore the extension if the filename already has one.
+        let len: u32 =
+            unsafe { SearchPathW(None, &filename, &extension_hstring, Some(&mut buffer), None) };
+
+        // If the function fails, the return value is zero.
+        if len == 0 {
+            continue;
+        }
+
+        // If the function succeeds, the value returned is the
+        // length of the string that is copied to the buffer,
+        // not including the terminating null character.
+        if len < buffer.len() as u32 {
+            buffer.truncate(len as usize);
+            return Ok(String::from_utf16(&buffer)?);
```

Comment:
FWIW you could also consider returning an `OsString` or even better `Path` here, as that would be consistent with other parts of Rust and this app. It would also allow you to revert the change to `absolute_path` below.